Helper/MerchantMetrics.php

```diff
@@ -0,0 +1,343 @@
+        // determines if we are in the Sandbox env or not
+        $base_uri =  $this->configHelper->isSandboxModeSet() ? 'https://api-sandbox.bolt.com/' : 'https://api.bolt.com/' ;
+
+        // Creates a Guzzle Client and Headers needed for Metrics Requests
+        return new \GuzzleHttp\Client(['base_uri' => $base_uri]);
+    }
+
+    /**
+     * Retrieves Headers needed to communicate with Bolt
+     *
+     * @return array
+     */
+    protected function setHeaders() {
+        return [
+            'Content-Type' => 'application/json',
+            'x-api-key' =>  $this->configHelper->getApiKey()
+        ];
+    }
+
+    /**
+     * Fetches the current root directory for magento and creates a bolt metrics file in that location
+     *
+     * @return string
+     */
+    protected function setFile() {
+        // determine root directory and add create a metrics file there
+        $objectManager = \Magento\Framework\App\ObjectManager::getInstance();
+        $directory = $objectManager->get('\Magento\Framework\Filesystem\DirectoryList');
+        $rootPath  =  $directory->getRoot();
+        return $rootPath . '/bolt_metrics.json';
+    }
+
+    /**
+     * Attempts to open a file and if it cannot open in 5 seconds it will return null
+     *
+     * @return Stream
+     */
+    public function waitForFile(){
+        $count = 0;
+        $timeoutMsecs = 10; //number of miliseconds of timeout (10 * 500) so 5 seconds
+
+        if ($this->metricsFile == null) {
+            $this->metricsFile = $this->setFile();
+        }
+
+        $workingFile = fopen($this->metricsFile, "a+");
+
+        // logic for properly grabbing file and locking it
+        while (!$this->lockFile($workingFile)) {
+            if ($count++ < $timeoutMsecs) {
+                usleep(500);
+            } else {
+                return null;
+                break;
+            }
+        }
+        return $workingFile;
+    }
+
+    /**
+     * Add a count metric to the array of metrics being stored
+     *
+     * @param string        $key    name of count metric
+     * @param int           $value  count hit 
+     *
+     * @return void
+     */
+    public function addCountMetric($key, $value)
+    {
+        $data = [
+                'value' => $value,
+                "metric_type" => "count",
+                "timestamp" => $this->getCurrentTime(),
+            ];
+        $this->metrics[$key] = $data;
+    }
+
+     /**
+     * Add a latency metric to the array of metrics being stored
+     *
+     * @param string        $key     name of latency metric
+     * @param int           $value  the total time of the metric
+     *
+     * @return void
+     */
+    public function addLatencyMetric($key, $value)
+    {
+        $data = [
+                'value' => $value,
+                "metric_type" => "latency",
+                "timestamp" => $this->getCurrentTime(),
+            ];
+        $this->metrics[$key] = $data;
+
+    }
```

Comment:
The two methods above can be easily joined into one, avoiding repeating the code.